dm/syncer/binlogstream/binlog_locations.go

```diff
@@ -190,34 +174,47 @@ func (l *locationRecorder) update(e *replication.BinlogEvent) {
 
 	switch ev := e.Event.(type) {
 	case *replication.GTIDEvent:
-		l.setCurEndGTID(e)
+		// following event should have new GTID set as end location
+		e2 := e
```

Comment:
Will be cause a problem if we don't define a new local variable and use `ev` instead?